scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,255 @@ output:
+scater::plotReducedDim(integrated_sce,
+                       dimred = "fastmnn_UMAP",
+                       # color each point by cell type
+                       colour_by = "celltype_broad",
+                       point_size= 0.5, 
+                       point_alpha = 0.4)
+```
+
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns.
+We see that there are at least 2 clear distinct groups for each cell type, 1 in each half of the UMAP.
+It looks like this is because the difference in diagnosis group accounts for differences across all cells in the sample, even those from the same type. 
+We can take advantage of another `ggplot` tool, the `facet_wrap()` function to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
+In the below plot we will color our cells by cell type but we will also use `facet_wrap()` to plot the cells from each subdiagnosis in their own separate panel of the plot.
+
+
+```{r celltype subdiagnosis UMAP}
+# # UMAP of all libraries, separating by diagnosis group and labeling cell type
 scater::plotReducedDim(integrated_sce,
                        dimred = "fastmnn_UMAP",
+                       # color each point by cell type
                        colour_by = "celltype_broad",
                        point_size= 0.5, 
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
+  # split plots by diagnosis group 
   facet_wrap(~ diagnosis_group)
 ```
-We can also look at the distribution of cell types across each sample.
 
-```{r}
+As expected, we see that the visual clusters of cell types are being separated most likely due to different subdiagnoses.
+Although we expect that the same cell types are present in both, each of the subdiagnoses for rhabdomyosarcoma have a distinct genomic and epigenomic landscape, likely accounting for the global gene expression changes seen between samples from these two subdiagnoses in the UMAP ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639), [Stewart _et al._](https://doi.org/10.1016/j.ccell.2018.07.012)).
+
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
+
+```{r celltype barplot}
+# filter coldata data.frame to only include tumor cells
 tumor_cells_df <- coldata_df %>%
   dplyr::filter(stringr::str_detect(celltype_broad, "Tumor"))
 
+# create a stacked barplot
 ggplot(tumor_cells_df, aes(x = sample, fill = celltype_broad)) + 
     geom_bar(color = "black", size = 0.2) +
     labs(
-      x = "Sample",
+      x = "Library",
       y = "Number of cells", 
       fill = "Cell type"
     ) +
   theme_bw() +
   theme(axis.text.x = element_text(angle = 90))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
   facet_wrap(~diagnosis_group, scales = "free_x")
```

Comment:
No, but there is one sample that doesn't have any cell types assigned so that sample won't show up here. It gets removed when we filter to only samples with tumor cells.

Reply:
Ah... that explains it. We should probably explicitly note that somewhere